bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -614,6 +624,10 @@ func (r *KubeadmConfigReconciler) joinControlplane(ctx context.Context, scope *S
 		conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesCorruptedReason, clusterv1.ConditionSeverityError, err.Error())
 		return ctrl.Result{}, err
 	}
+
+	// Update the certificate expiration time in the config.
+	r.addCertificateExpiryAnnotation(scope.Config)
```

Comment:
I assume those are the right places to add the annotation, but just to double check:
* during regular reconcile this func is only called initially on first reconcile of KubeadmConfig?
* for KubeadmConfigs which existed before this feature was introduced the annotation would not be added?